### PR TITLE
Reset password service

### PR DESCRIPTION
**Pull request by sterlingwellscaffeine**
1.) Adds the following methods to the auth service
  - sendPasswordResetEmail(email:string) - sends a request to the server to send the password reset email
 - resetPassword(token:string, userId: string) - sends a request to the server to reset the password
 
2.) Adds the correct mat-error modal to the login-screen when the server fails to send the reset password email

3.) Adds unit tests that show that the MatDialog is correct triggered upon the clicking of the Forgot Password button

4.)

### CODE REVIEW

File: src/interface/src/app/login/login.component.ts [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 import { MatDialog } from '@angular/material/dialog';
-import { ResetPasswordDialog } from './reset_password_dialog';
+
+import * as signInMessages from '../shared/constants';
+import { ResetPasswordDialog } from './reset-password-dialog/reset_password_dialog';
 
 @Component({
   selector: 'app-login',
-  templateUrl: './login.component.html',
+  templateUrl: './login.component.ng.html',
```
Comment (runforyourlives) [inline src/interface/src/app/login/login.component.ts:13]: is this file missing from this PR?  I don't see it in the codebase.

File: src/interface/src/app/shared/nav-bar/nav-bar.component.spec.ts [hunk 2]
```diff
@@ -12,7 +11,6 @@ describe('NavBarComponent', () => {
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       declarations: [NavBarComponent],
-      schemas: [CUSTOM_ELEMENTS_SCHEMA],
```
Comment (runforyourlives) [inline src/interface/src/app/shared/nav-bar/nav-bar.component.spec.ts:15]: FYI - I can't recall what will happen here, but it was added very recently as part of nav bar work that was being done (which requires a flag that you may not have enabled).  Is there anything unwanted if you leave CUSTOM_ELEMENTS_SCHEMA in?

File: src/interface/src/app/login/login.component.ts (hunk 1 above)
Comment (sterlingwellscaffeine) [reply src/interface/src/app/login/login.component.ts]: So I was fixing the naming convention based on a comment made by Maciek and so this comment I think is based on an outdated piece of code.

File: src/interface/src/app/shared/nav-bar/nav-bar.component.spec.ts (hunk 2 above)
Comment (sterlingwellscaffeine) [reply src/interface/src/app/shared/nav-bar/nav-bar.component.spec.ts]: Git erroneously marked part of this code as a merge conflict so I believe I resolved it incorrectly. It should be good to go now.

**Review by runforyourlives — APPROVED** [review]

**Pull request merged**